integration_tests/tests/end_to_end.rs

```diff
@@ -613,20 +639,125 @@ async fn test_kafka() {
                 </station>
             </KvalobsData>"#;
 
-        kvkafka::process_message(
-            &mut open_conn,
-            &mut restricted_conn,
-            kafka_xml,
-            mock_permit_tables(),
-        )
-        .await
-        .unwrap();
+        producer
+            .send_result(FutureRecord::to(KAFKA_TOPIC).key("").payload(kafka_xml))
+            .unwrap()
+            .await
+            .unwrap()
+            .unwrap();
 
         // TODO: we do not have an API endpoint to query the flags.kvdata table
-        assert!(open_conn
-            .query_one("SELECT * FROM legacy.data", &[])
-            .await
-            .is_ok());
+        let open_conn = db_pools.open.get().await.unwrap();
+
+        // As we have no way to sync with message processing in kvkafka ingestion, we just keep
+        // trying to fetch data with a timeout
+        let timeout = std::time::Duration::from_secs(10);
+        let timeout_start = Instant::now();
+        loop {
```

Comment:
Are we expecting for it to take 10 seconds to ingest a single message?? I honestly would just drop to 1 second and remove the loop, because if it takes longer there's definitely something wrong with the ingestion :sweat_smile: (I know it might be flaky, but still...)

Reply:
Weird, on my machine it "consistently" takes 0.3 s :thinking: maybe it's cached?

Reply:
?

Reply:
I had commented out the `loop` (and added a `tokio::time::sleep` instead of the timeout), but without changing the `if let Ok(data_row) = ...` :sweat_smile: